pkg/system/phase4_configuring.go

```diff
@@ -537,16 +537,20 @@ func (r *Reconciler) awaitEndpointDeploymentPods() error {
 func (r *Reconciler) ReconcileHPAEndpoint() error {
 	// Wait for the the endpoint deployment to become ready
 	// only if HPA was not created yet
+
 	if r.HPAEndpoint.UID == "" {
 		if err := r.awaitEndpointDeploymentPods(); err != nil {
 			return err
 		}
 	}
-
-	if err := r.ReconcileObject(r.HPAEndpoint, r.SetDesiredHPAEndpoint); err != nil {
+	if err := r.reconcileAutoscaler(); err != nil {
 		return err
 	}
+	return r.updateNoobaaEndpoint()
+
+}
 
+func (r *Reconciler) updateNoobaaEndpoint() error {
 	max := r.HPAEndpoint.Spec.MaxReplicas
```

Comment:
This code seems irrelevant now, since `HPAEndpoint` is only relevant for HPAV1. maybe we should get min\max from `endpointsSpec`?